lib/common/client/base.rb

```diff
@@ -14,10 +14,20 @@ class Base
       include SentryLogging
 
       class << self
+        attr_accessor :service_exception
+
         def configuration(configuration = nil)
           @configuration ||= configuration.instance
         end
+
+        def use_service_exception(exception)
+          unless exception.ancestors.include?(Common::Exceptions::BackendServiceException)
+            raise "[#{exception}] must inherit from BackendServiceException"
+          end
+          @service_exception = exception
+        end
       end
+      delegate :service_exception, to: 'self.class'
```

Comment:
I think a better home for this would be in Common::Configuration, since that is where we typically define configuration options. Additionally since the service_name is there, it would be ideal if this error class could live there as well. I'm not so sure the `unless` block will be necessary, it's trivial to generate a class based on the service_name that is already defined. The class will automatically be generated when Common::Configuration is initialized, and since Common::Configuration is a singleton, it's really a perfect place to this (no risk of initializing the constant twice).

Reply:
something like:

```
Object.const_set("#{service_name}ServiceException", Class.new(BackendServiceException))
```

would dynamically generate the class based on service_name when configuration is initialized, and then Common::Client would define service_exception as always being:

"#{self.config.service_name}ServiceException".constantize || BackendServiceException